### ISSUE TITLE
6.2.4: sphinx warnings

### ISSUE THREAD

**Issue by kloczek** [pytest-dev/pytest#8730]
There are some sphinx documentation minor issues:
```console
+ /usr/bin/python3 setup.py build_sphinx -b man
running build_sphinx
Running Sphinx v4.0.2
making output directory... done
[autosummary] generating autosummary for: adopt.rst, announce/index.rst, announce/release-2.0.0.rst, announce/release-2.0.1.rst, announce/release-2.0.2.rst, announce/release-2.0.3.rst, announce/release-2.1.0.rst, announce/release-2.1.1.rst, announce/release-2.1.2.rst, announce/release-2.1.3.rst, ..., sponsor.rst, talks.rst, tidelift.rst, tmpdir.rst, unittest.rst, usage.rst, warnings.rst, writing_plugins.rst, xunit_setup.rst, yieldfixture.rst
loading intersphinx inventory from https://pluggy.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
building [mo]: all of 0 po files
building [man]: all source files
updating environment: [new config] 198 added, 0 changed, 0 removed
reading sources... [100%] yieldfixture
/usr/lib/python3.8/site-packages/_pytest/pytester.py:docstring of _pytest.pytester.Pytester.TimeoutExpired:1: WARNING: duplicate object description of _pytest.pytester.Pytester.TimeoutExpired, other instance in reference, use :noindex: for one of them
/usr/lib/python3.8/site-packages/_pytest/main.py:docstring of _pytest.main.Session:1: WARNING: duplicate object description of _pytest.main.Session, other instance in reference, use :noindex: for one of them
/usr/lib/python3.8/site-packages/_pytest/main.py:docstring of _pytest.main.Failed:1: WARNING: duplicate object description of _pytest.main.Failed, other instance in reference, use :noindex: for one of them
docstring of builtins.Interrupted:1: WARNING: duplicate object description of builtins.Interrupted, other instance in reference, use :noindex: for one of them
/usr/lib/python3.8/site-packages/_pytest/main.py:docstring of _pytest.main.Session:1: WARNING: duplicate object description of _pytest.main.Session, other instance in reference, use :noindex: for one of them
docstring of builtins.Interrupted:1: WARNING: duplicate object description of builtins.Interrupted, other instance in reference, use :noindex: for one of them
/usr/lib/python3.8/site-packages/_pytest/main.py:docstring of _pytest.main.Failed:1: WARNING: duplicate object description of _pytest.main.Failed, other instance in reference, use :noindex: for one of them
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... pytest.1 { } python-pytest.3 { } done
build succeeded, 7 warnings.
```
- [x] a detailed description of the bug or problem you are having
- [x] output of `pip list` from the virtual environment you are using
- [x] pytest and operating system versions
- [x] minimal example if possible

pytest 6.2.4
Linux x86_64
```
$ pip list
Package                       Version
----------------------------- -----------------
aiofiles                      0.6.0
aiohttp                       3.7.4.post0
aiohttp-cors                  0.7.0
alabaster                     0.7.12
amqp                          5.0.6
ansi                          0.2.0
anytree                       2.8.0
apipkg                        1.5
appdirs                       1.4.4
argcomplete                   1.11.1
Arpeggio                      1.10.2
arrow                         1.1.0
asgiref                       3.3.4
asn1crypto                    1.4.0
aspectlib                     1.5.2
astor                         0.8.1
astroid                       2.5.7
async-generator               1.10
async-timeout                 3.0.1
atomicwrites                  1.3.0
attrs                         21.2.0
autoapi                       2.0.1
autodocsumm                   0.2.4
Automat                       20.2.0
Babel                         2.9.1
backcall                      0.2.0
bcrypt                        3.2.0
Beaker                        1.10.0
beautifulsoup4                4.9.3
betamax                       0.8.1
billiard                      3.6.4.0
blinker                       1.4
blivet                        3.4.0
botocore                      1.20.86
bottle                        0.12.19
breathe                       4.30.0
Brlapi                        0.8.0
Brotli                        1.0.7
build                         0.4.0
cachetools                    4.2.2
cairocffi                     1.2.0
CairoSVG                      2.5.2
case                          1.5.3
cbor2                         5.2.0
CCColUtils                    1.5
certifi                       2018.10.15
cffi                          1.14.5
cfgv                          3.2.0
chai                          1.1.2
changelog                     0.5.5
chardet                       3.0.4
click                         8.0.1
click-default-group           1.2.2
colorama                      0.4.4
colorlog                      4.8.0
commonmark                    0.9.0
constantly                    15.1.0
contextvars                   2.4
cov-core                      1.15.0
coverage                      5.6b1
cram                          0.7
cryptography                  3.3.1
cryptography-vectors          3.4.7
cssselect                     1.1.0
cssselect2                    0.4.1
curio                         1.5
cycler                        0.10.0
Cython                        0.29.22
dateparser                    0.7.4
dateutils                     0.6.12
dbus-python                   1.2.16
ddt                           1.4.2
decopatch                     1.4.8
decorator                     5.0.8
defusedxml                    0.7.1
distlib                       0.3.1
distro                        1.5.0
Django                        3.2.2
dnspython                     2.1.0
docopt                        0.6.2
docutils                      0.16
docutils-stubs                0.0.21
dogtail                       0.9.11
dpkt                          1.9.6
dulwich                       0.20.23
editdistance                  0.5.3
editdistance-s                1.0.0
elasticsearch                 7.11.0
elementpath                   2.2.2
entrypoints                   0.3
evdev                         1.4.0
exam                          0.10.6
execnet                       1.7.1
extras                        1.0.0
factory-boy                   3.2.0
Faker                         8.4.0
fido2                         0.9.1
fields                        5.0.0
filelock                      3.0.12
fixtures                      3.0.0
flake8                        3.9.1
Flask                         1.1.2
Flask-SQLAlchemy              2.5.1
fluidity-sm                   0.2.0
fonttools                     4.24.4
freezegun                     1.0.0
func-timeout                  4.4.0
future                        0.18.2
fuzzywuzzy                    0.18.0
gcovr                         4.2
Genshi                        0.7.5
genty                         1.3.2
gevent                        21.1.2
geventhttpclient              1.4.5
gi-docgen                     2021.1
gitdb                         4.0.7
GitPython                     3.1.14
gpg                           1.15.1-unknown
greenlet                      1.1.0
guzzle-sphinx-theme           0.7.11
h11                           0.12.0
h2                            3.2.0
hacking                       4.0.0
hidapi                        0.10.1
hpack                         3.0.0
html5lib                      1.1
httmock                       1.4.0
httpbin                       0.7.0
humanize                      0.5.1
hyperframe                    5.2.0
hyperlink                     21.0.0
hypothesis                    6.13.12
identify                      2.2.8
idna                          3.1
imagesize                     1.2.0
immutables                    0.15
importlib-metadata            4.4.0
incremental                   17.5.0
inflect                       5.3.0
iniconfig                     1.1.1
invoke                        1.5.0
ipython-genutils              0.2.0
iso8601                       0.1.14
isodate                       0.6.0
isort                         5.7.0
itsdangerous                  2.0.1
jaraco.classes                3.2.1
jaraco.envs                   0.0.0
jaraco.functools              3.3.0
jaraco.itertools              6.0.1
jaraco.packaging              8.2.1
jaraco.path                   0.0.0
jaraco.text                   3.5.0
javapackages                  5.3.0
jedi                          0.18.0
Jinja2                        3.0.1
jmespath                      0.10.0
jsonschema                    3.2.0
kiwisolver                    1.2.0
lark-parser                   0.11.2
lazy-object-proxy             1.6.0
lexicon                       1.0.0
libcomps                      0.1.15
libevdev                      0.9
linecache2                    1.0.0
lit                           12.0.0.dev0
lockfile                      0.12.2
louis                         3.17.0
lxml                          4.6.3
makefun                       1.11.3
Mako                          1.1.4.dev0
mallard-ducktype              1.0.2
marisa                        0.0.0
Markdown                      3.3.4
MarkupSafe                    2.0.1
mccabe                        0.6.1
meson                         0.58.0
metaextract                   1.0.7
mistune                       2.0.0a6
mock                          4.0.3
more-itertools                8.8.0
msgpack                       1.0.2
mujson                        1.4
multidict                     5.1.0
mypy                          0.812
mypy-extensions               0.4.3
netaddr                       0.8.0
nftables                      0.1
nocasedict                    1.0.2
nocaselist                    1.0.4
nodeenv                       1.6.0
nose                          1.3.7
nose2                         0.10.0
nox                           2020.12.31
ntplib                        0.4.0
numpy                         1.18.2
olefile                       0.46
openidc-client                0.6.0
openstackdocstheme            2.2.7
ordered-set                   4.0.2
outcome                       1.1.0
packaging                     20.9
Pallets-Sphinx-Themes         2.0.1
pandas                        1.2.4
parameterized                 0.8.1
paramiko                      2.7.2
parso                         0.8.0
parver                        0.3.1
Paste                         3.5.0
PasteDeploy                   2.1.1
path                          15.1.2
pathspec                      0.8.1
pbr                           5.4.3
pcapy                         0.11.4
pep517                        0.10.0
pexpect                       4.8.0
pickleshare                   0.7.5
pid                           3.0.4
Pillow                        7.1.1
pip                           20.0.2
pluggy                        0.13.1
ply                           3.11
pocketlint                    0.21
poetry-core                   1.0.3
polib                         1.1.1
pretend                       1.0.9
prettytable                   0.7.2
priority                      1.3.0
process-tests                 2.1.1
productmd                     1.33
prompt-toolkit                3.0.18
psutil                        5.6.7
ptyprocess                    0.7.0
purl                          1.5
pwquality                     1.4.4
py                            1.10.0
py-cpuinfo                    8.0.0
py2pack                       0.8.6
pyasn1                        0.4.8
pyasn1-modules                0.2.8
pybind11                      2.6.2
pycairo                       1.18.2
pycodestyle                   2.7.0
pycparser                     2.19
pycrypto                      2.6.1
pycups                        1.9.74
pycurl                        7.43.0.5
pydantic                      1.8.2
pydocstyle                    6.0.0
pyenchant                     3.2.0
pyfakefs                      4.4.0
pyflakes                      2.3.1
pygal                         3.0.0.dev1
Pygments                      2.9.0.dev20210503
pygments-ansi-color           0.0.6
pygments-pytest               2.2.0
PyGObject                     3.40.1
PyHamcrest                    2.0.2
pyinotify                     0.9.6
pykickstart                   3.32
pylast                        3.3.0
pylint                        2.7.4
pylons-sphinx-themes          1.0.13
PyNaCl                        1.4.0
pyOpenSSL                     20.0.1
pyparsing                     2.4.6
pyparted                      3.11.5
PyQt5                         5.14.2
PyQt5-sip                     4.19.22
pyquery                       1.4.4.dev0
pyrad                         2.4
pyrsistent                    0.17.3
pyscard                       2.0.0
pyScss                        1.3.7
pyserial                      3.4
PySocks                       1.7.1
pytest                        6.2.4
pytest-aiohttp                0.3.0
pytest-asyncio                0.15.1
pytest-cases                  3.4.6
pytest-checkdocs              2.7.0
pytest-cov                    2.12.1
pytest-expect                 1.1.0
pytest-fixture-config         1.7.0
pytest-flake8                 1.0.7
pytest-forked                 1.3.0
pytest-freezegun              0.4.2
pytest-httpbin                1.0.0
pytest-isort                  1.3.0
pytest-mock                   3.6.1
pytest-randomly               3.8.0
pytest-rerunfailures          9.1.1
pytest-runner                 5.3.1
pytest-shutil                 1.7.0
pytest-timeout                1.4.2
pytest-toolbox                0.5
pytest-virtualenv             1.7.0
pytest-xdist                  2.2.1
pytest-xprocess               0.17.1
python-augeas                 0.5.0
python-dateutil               2.8.1
python-dbusmock               0.22.0
python-mimeparse              1.6.0
python-subunit                1.4.0
python-xlib                   0.30
pytz                          2021.1
pyudev                        0.22.0
pyusb                         1.1.1
PyUtilib                      6.0.0
pywbem                        1.1.3
pyxattr                       0.7.2
pyxdg                         0.26
PyYAML                        5.4.1
raven                         6.10.0
rdflib                        5.0.0
re-assert                     1.1.0
readthedocs-sphinx-ext        2.1.4
recommonmark                  0.7.1
redis                         3.5.3
regex                         2021.4.4
releases                      1.6.3
reno                          3.2.0
requests                      2.25.1
requests-download             0.1.2
requests-file                 1.5.1
requests-ftp                  0.3.1
requests-mock                 1.9.3
requests-toolbelt             0.9.1
requests-unixsocket           0.2.0
responses                     0.12.1
rpm                           4.16.1.3
rsa                           4.7.2
rst.linker                    2.2.0
s3transfer                    0.4.2
scons                         3.1.2
scour                         0.38.2
selenium                      3.14.0
selinux                       3.2
semantic-version              2.8.5
service-identity              21.1.0
setools                       4.3.0
setuptools                    46.1.3
setuptools-git                1.2
setuptools-scm                6.0.1
should-dsl                    2.1.2
simplejson                    3.17.2
simpleline                    1.8
six                           1.16.0
slip                          0.6.5
slip.dbus                     0.6.5
smartypants                   2.0.1
smmap                         4.0.0
sniffio                       1.2.0
snowballstemmer               2.1.0
solaar                        1.0.6
sortedcontainers              2.1.0
sos                           4.1
soupsieve                     2.2.1
spec                          1.4.1
Sphinx                        4.0.2
sphinx-argparse-cli           0.0.0
sphinx-autodoc-typehints      0.0.0
sphinx-celery                 2.0.0
sphinx-gallery                0.8.2
sphinx-hoverxref              0.5b1
sphinx-issues                 1.2.0
sphinx-lv2-theme              1.0.0
sphinx-markdown-tables        0.0.15
sphinx-paramlinks             0.5.1
sphinx-prompt                 1.4.0
sphinx-py3doc-enhanced-theme  2.3.2
sphinx-removed-in             0.2.1
sphinx-rst-builder            0.0.3
sphinx-rtd-theme              0.4.3
sphinx-selective-exclude      1.0.3
sphinx-tabs                   3.0.0
sphinx-typlog-theme           0.8.0
sphinxcontrib-apidoc          0.3.0
sphinxcontrib-applehelp       1.0.1
sphinxcontrib-asyncio         0.3.0
sphinxcontrib-autoprogram     0.1.7
sphinxcontrib-devhelp         1.0.1
sphinxcontrib-htmlhelp        1.0.1
sphinxcontrib-httpdomain      1.7.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-log-cabinet     1.0.1
sphinxcontrib-qthelp          1.0.2
sphinxcontrib-restbuilder     0.3
sphinxcontrib-serializinghtml 1.1.3
sphinxcontrib-spelling        7.2.1
sphinxcontrib-trio            1.1.2
sphinxcontrib-websupport      1.2.4
sphobjinv                     2.0.1
SQLAlchemy                    1.4.17
sqlparse                      0.4.2.dev0
stdio-mgr                     1.0.1
stdlib-list                   0.6.0
stone                         3.2.1
sure                          1.4.11
sybil                         2.0.1
systemd-python                234
Tempita                       0.5.2
termcolor                     1.1.0
testpath                      0.4.4
testrepository                0.0.20
testresources                 2.0.1
testscenarios                 0.5.0
testtools                     2.4.0
text-unidecode                1.3
texttable                     1.6.3
tinycss2                      1.1.0
toml                          0.10.2
tomlkit                       0.7.2
tornado                       6.1
towncrier                     19.9.0
tox                           3.14.6
tpm2-pkcs11-tools             1.33.7
traitlets                     5.0.5
trio                          0.18.0+dev
trustme                       0.7.0
Twisted                       21.2.0
typed-ast                     1.4.3
typing-extensions             3.10.0.0
typogrify                     2.0.7
tzlocal                       2.1
u-msgpack-python              2.5.2
ujson                         4.0.2
unittest-mixins               1.6
unittest2                     1.1.0
urllib3                       1.26.5
userpath                      1.6.0
uTidylib                      0.7
uvicorn                       0.13.4
validators                    0.18.1
vine                          5.0.0
virtualenv                    20.4.7
waitress                      1.4.4
watchgod                      0.7
wcwidth                       0.2.5
webcolors                     1.11.1
webencodings                  0.5.1
WebOb                         1.8.7
websocket-client              1.0.1
websockets                    9.0.1
WebTest                       2.0.35
Werkzeug                      0.16.0
wheel                         0.36.2
Whoosh                        2.7.4
wikipedia                     1.4.5
wrapt                         1.12.1
WSGIProxy2                    0.4.6
xcffib                        0.11.1
xmlschema                     1.6.2
xmltodict                     0.12.0
yamlloader                    1.0.0
yarl                          1.6.3
youtube-dl                    2021.5.16
zipp                          3.4.1
zope.event                    4.2.0
zope.interface                5.4.0
zope.testing                  4.7
zzzeeksphinx                  1.2.3
```

**Comment by nicoddemus** [comment]
Hi @kloczek,

Thanks for the report!

I gave this a shot, the problem is some aliases we have.

About `TimeoutExpired`, we declare it in `Pytester`:

```python
@final
class Pytester:
    ...

    class TimeoutExpired(Exception):
        pass
```

And declare an alias in `Testdir`:

```python
@final
@attr.s(repr=False, str=False, init=False)
class Testdir:
    ...
    TimeoutExpired: "Final" = Pytester.TimeoutExpired
```

As a test, commenting out the `TimeoutExpired` line from `Testdir` makes the warning go away, however I don't know how to mark `Testdir.TimeoutExpired` as `:noindex:` like the warning suggests. I've tried this:

```python
    TimeoutExpired: "Final" = Pytester.TimeoutExpired  #: :meta noindex:
```

As I've seen in the docs, but that did not work.

Suggestions here are welcome!

**Comment by kloczek** [comment]
Looks like in just released 7.1.2 there are still some warnings:
```console
+ /usr/bin/sphinx-build -n -T -b man doc/en build/sphinx/man
Running Sphinx v4.5.0
making output directory... done
[autosummary] generating autosummary for: adopt.rst, announce/index.rst, announce/release-2.0.0.rst, announce/release-2.0.1.rst, announce/release-2.0.2.rst, announce/release-2.0.3.rst, announce/release-2.1.0.rst, announce/release-2.1.1.rst, announce/release-2.1.2.rst, announce/release-2.1.3.rst, ..., reference/customize.rst, reference/exit-codes.rst, reference/fixtures.rst, reference/index.rst, reference/plugin_list.rst, reference/reference.rst, sponsor.rst, talks.rst, tidelift.rst, yieldfixture.rst
loading intersphinx inventory from https://pluggy.readthedocs.io/en/stable/objects.inv...
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://numpy.org/doc/stable/objects.inv...
loading intersphinx inventory from https://pip.pypa.io/en/stable/objects.inv...
loading intersphinx inventory from https://tox.wiki/en/stable/objects.inv...
loading intersphinx inventory from https://virtualenv.pypa.io/en/stable/objects.inv...
loading intersphinx inventory from https://setuptools.pypa.io/en/stable/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: [new config] 216 added, 0 changed, 0 removed
reading sources... [100%] yieldfixture
/home/tkloczko/rpmbuild/BUILD/pytest-7.1.2/src/_pytest/pytester.py:docstring of _pytest.pytester.Pytester.TimeoutExpired:1: WARNING: duplicate object description of _pytest.pytester.Pytester.TimeoutExpired, other instance in reference/reference, use :noindex: for one of them
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... WARNING: "man_pages" config value references unknown document usage
python-pytest.3 { } done
build succeeded, 2 warnings.
```
Nevertheless it is good progress 😄

**Comment by bluetech** [comment]
I think this is fixed
